docs/providers/aws/events/alb.md

```diff
@@ -195,3 +195,53 @@ functions:
           conditions:
             path: /hello
 ```
+
+## Configuring Health Checks
+
+Health checks for target groups with a _lambda_ target type are disabled by default.
+
+To simply enable the health check on a target group associated with an alb event, set the alb event's `healthCheck` property to `true`.
+
+```yml
+functions:
+  albEventConsumer:
+    handler: handler.hello
+    events:
+      - alb:
+          listenerArn: arn:aws:elasticloadbalancing:us-east-1:12345:listener/app/my-load-balancer/50dc6c495c0c9188/
+          priority: 1
+          conditions:
+            path: /hello
+          healthCheck: true
+```
+
+Once enabled, a target group health check can be disabled by explicitly setting the alb event's `healthCheck` setting to `false`.
```

Comment:
Is this documentation line needed?

Wouldn't `healthCheck` be naturally disabled with a deployment when user just removes this setting

Reply:
I tested this manually and confirmed that when the health check settings are removed from the template of an existing CloudFormation stack, the target group is not modified. Those settings only seem to change for an existing resource when set explicitly.

We can set `HeatlhCheckEnabled` to false by default if a `healthCheck` setting is not provided, but I wasn't willing to be that opinionated without further discussion.

Reply:
> I tested this manually and confirmed that when the health check settings are removed from the template of an existing CloudFormation stack, the target group is not modified. 

That looks as unfortunate bug on CloudFormation side

> We can set HeatlhCheckEnabled to false by default if a healthCheck setting is not provided, but I wasn't willing to be that opinionated without further discussion.

Yes, I also believe it's best to not apply setting at all if it's not set (and it's what works in most case). Still here, due to how it's handled on CloudFormation side, we may need to apply an exception and set it to`false` by default  (to ensure it's removed when user doesn't set this).

Is setting this to `false` equivalent to not setting it at all for fresh deployments?

Reply:
Yes, setting healthCheck to false would be the equivalent of not setting HealthCheckEnabled at all  for new deployments. The current documented behavior for Lambda type target groups is that health checks are disabled by default.

Reply:
Therefore I'd say let's set `enabled: false` by default (so always provide `enabled` setting)

Reply:
Would that mean forcibly setting the remaining health check settings back to their defaults as well or do we only want to override CloudFormation's behavior for `HealthCheckEnabled`?

Reply:
@DavidSeptimus I believe that when health check is disabled, then there's no need to keep any other settings at defaults, as anyway they're not effective in any way, right?

Still, question is, what happens if we first enable health check with custom settings, then disable it, and then enable it just via `healthCheck: true`.  Technically generated CloudFormation should list only customized properties. Will they then effectively act at their defaults? Or will some of them now be effective with custom values set when health check was first enabled?

If latter is the case, then when health check is enabled, I think we need to also force those settings (which do not reset on its own) to default values specifically.

Reply:
> I believe that when health check is disabled, then there's no need to keep any other settings at defaults, as anyway they're not effective in any way, right?

That's correct.

> Still, question is, what happens if we first enable health check with custom settings, then disable it, and then enable it just via healthCheck: true. Technically generated CloudFormation should list only customized properties. Will they then effectively act at their defaults? Or will some of them now be effective with custom values set when health check was first enabled?

The additional health check settings will use the original custom values unless we force them back to defaults.

> If latter is the case, then when health check is enabled, I think we need to also force those settings (which do not reset on its own) to default values specifically.

That's my instinct as well. The only potential concern I see there is potential drift from AWS's documented defaults if they decide to change them in the future.

Reply:
> The only potential concern I see there is potential drift from AWS's documented defaults if they decide to change them in the future.

True, but I think that'll be unlikely, I believe it'll be also a breaking change on their side which they should signal with new API version

So when not set, I would just stick to `enabled: false` and when set I'd enforce defaults for all properties for which previous customization may leak

Reply:
I'm comfortable with that. The discussed changes are included in my latest commit.